File: src/main/java/io/github/mg138/tsbook/utils/config/AbstractConfig.java
package io.github.mg138.tsbook.utils.config;

import io.github.mg138.tsbook.utils.Translate;
import org.bukkit.configuration.file.YamlConfiguration;
import org.bukkit.plugin.java.JavaPlugin;

import java.io.File;

public abstract class AbstractConfig {
    public YamlConfiguration config;
    public JavaPlugin plugin;
    public File JAR;
    public Translate translate;

    public abstract void load(JavaPlugin p, File j);

    public abstract void unload();
}

File: src/main/java/io/github/mg138/tsbook/setting/AbstractConfig.kt
package io.github.mg138.tsbook.setting

import io.github.mg138.tsbook.utils.Translate
import org.bukkit.plugin.java.JavaPlugin
import java.io.File

abstract class AbstractConfig {
    abstract var plugin: JavaPlugin
    abstract var jar: File
    abstract var translate: Translate
    abstract fun load(plugin: JavaPlugin, jar: File)
    abstract fun unload()
}